test/minitest/test_minitest_mock.rb

```diff
@@ -275,6 +275,18 @@ def test_mock_block_is_passed_function_params
     assert_mock mock
   end
 
+  def test_mock_block_is_passed_keyword_args
+    arg1, arg2, arg3 = :bar, [1, 2, 3], { :a => "a" }
+    mock = Minitest::Mock.new
+    mock.expect :foo, nil do |k1:, k2:, k3:|
+      k1 == arg1 && k2 == arg2 && k3 == arg3
+    end
+
+    mock.foo(k1: arg1, k2: arg2, k3: arg3)
+
+    assert_mock mock
+  end
```

Comment:
I've adopted this test and expanded it into 9 edge cases (block vs args, missing vs extra, bad keys, bad vals) covering what I think is the whole gamut. Thank you for the nudge!